client/src/components/pages/database/license-availability-table.tsx

```diff
@@ -1,7 +1,30 @@
-import mockAvailabilityData from './mock-availability-data';
+import licenseData from '../../../data/licenses.json';
+import { eligibleBostonZipcodes } from '@/services/data-interface/data-interface';
 import CustomTable from "@components/ui/table";
+import {useState, useEffect} from 'react';
+import { validateBusinessLicense, getAvailableLicensesByZipcode, BusinessLicense, EligibleBostonZipcode } from '@/services/data-interface/data-interface';
+import { MAX_ALL_ALC_PER_ZIP, MAX_AVAILABLE_PER_ZIP, MAX_BEER_WINE_PER_ZIP } from '@/services/data-interface/data-interface';
+import { RowWithSubRows } from '@components/ui/table';
 
```

Comment:
See comment below for reasons for moving this up here.

(Also adjust intentation)
```suggestion


  const formatData = (data: BusinessLicense[], zipcodeList: Set<EligibleBostonZipcode>) => {
      const zips = [...zipcodeList]
      const d = zips.map((zipcode) => {
       const {totalAvailable, allAlcoholAvailable, beerWineAvailable} = getAvailableLicensesByZipcode(data, zipcode); 
       const entry = {
           rowData: [zipcode, String(totalAvailable), '-', String(MAX_AVAILABLE_PER_ZIP - totalAvailable), String(MAX_AVAILABLE_PER_ZIP)],
           subRowData: [
             ['All Alcohol Licenses', String(allAlcoholAvailable), '-', String(MAX_ALL_ALC_PER_ZIP - allAlcoholAvailable), String(MAX_ALL_ALC_PER_ZIP)],
             ['Beer & Wine Licenses', String(beerWineAvailable), '-', String(MAX_BEER_WINE_PER_ZIP - beerWineAvailable), String(MAX_BEER_WINE_PER_ZIP)],
           ] 
       }
       return entry as RowWithSubRows;
      })
      return d;
  }
```